tests/generic_config_updater/test_multiasic_addcluster.py

```diff
@@ -0,0 +1,169 @@
+    local_minigraph = os.path.join(local_dir, f"{duthost.hostname}-minigraph.xml")
+    duthost.fetch(src=MINIGRAPH, dest=local_minigraph, flat=True)
+    refactor = MinigraphRefactor(TARGET_LEAF)
+    refactor.process_minigraph(local_minigraph, local_minigraph)
+    duthost.copy(src=local_minigraph, dest=MINIGRAPH)
+
+    # Step 3: Reload minigraph
+    logger.info("Reloading minigraph using 'config load_minigraph -y'")
+    duthost.shell("sudo config load_minigraph -y", module_ignore_errors=False)
+
+    # Step 3.1 check corresponding tables in CONFIG_DB, if not, create them by mini patch.
+    check_tables = ["BGP_NEIGHBOR", "PFC_WD", "CABLE_LENGTH", "BUFFER_PG", "PORT_QOS_MAP", "DEVICE_NEIGHBOR_METADATA"]
+    logger.info(f"Checking and creating tables: {check_tables}")
+    create_table_if_not_exist(duthost, check_tables)
+
+    # Step 4: Apply addcluster.json
+    logger.info("Applying addcluster.json patch")
+    with open(ADDCLUSTER_FILE) as file:
+        json_patch = json.load(file)
+    tmpfile = generate_tmpfile(duthost)
+    try:
+        apply_patch_result = apply_patch(duthost, json_data=json_patch, dest_file=tmpfile)
+        if apply_patch_result['rc'] != 0 or "Patch applied successfully" not in apply_patch_result['stdout']:
+            pytest.fail(f"Failed to apply patch: {apply_patch_result['stdout']}")
+    finally:
+        delete_tmpfile(duthost, tmpfile)
+
+    # Step 5: Check port status
+    for port in ["Ethernet16", "Ethernet24"]:
+        result = duthost.shell(f"show interface status {port}", module_ignore_errors=False)["stdout"]
+        pytest_assert("up" in result, f"{port} is not up")
+
+    # Step 6: Check PortChannel exists
+    result = duthost.shell("show interfaces portchannel", module_ignore_errors=False)["stdout"]
+    pytest_assert("PortChannel101" in result, "PortChannel101 not found in portchannel list")
+
+    # Step 7: Check BGP session
+    result = duthost.shell("show ip bgp summary", module_ignore_errors=False)["stdout"]
+    pytest_assert("10.0.0.13" in result and "Estab" in result,
+                  "BGP session with 10.0.0.13 not established")
+
+    # Step 8: Verify all addcluster.json changes are reflected in CONFIG_DB
+    keys_to_check = [
+        # PORT entries
+        "PORT|Ethernet16",
+        "PORT|Ethernet24",
+
+        # PortChannel entries
+        "PORTCHANNEL|PortChannel101",
+        "PORTCHANNEL_MEMBER|PortChannel101|Ethernet16",
+        "PORTCHANNEL_MEMBER|PortChannel101|Ethernet24",
+        "PORTCHANNEL_INTERFACE|PortChannel101",
+        "PORTCHANNEL_INTERFACE|PortChannel101|10.0.0.12/31",
+        "PORTCHANNEL_INTERFACE|PortChannel101|fc00::19/126",
+
+        # BGP neighbors
+        "BGP_NEIGHBOR|10.0.0.13",
+        "BGP_NEIGHBOR|fc00::1a",
+
+        # ACL table entries
+        "ACL_TABLE|EVERFLOW",
+        "ACL_TABLE|EVERFLOWV6",
+
+        # DEVICE_NEIGHBOR_METADATA entries
+        "DEVICE_NEIGHBOR_METADATA|Ethernet3~11",
```

Comment:
>~1 [](http://example.com/codeflow?start=43&length=2)

seems you manually escaped string. Could you use human readable string, and use json patch escape function to handle escaping?